core/debug/src/debug.c

```diff
@@ -230,3 +230,117 @@ void debug_fault(THaltError reason, uint32_t lr, uint32_t sp)
     debug_mpu_config();
     DEBUG_PRINT_END();
 }
+
+static void __debug_reboot(void)
+{
+    UVISOR_SVC(UVISOR_SVC_ID_DEBUG_REBOOT, "");
+}
+
+void debug_reboot(void)
+{
+    if (!g_debug_box.initialized || g_active_box != g_debug_box.box_id) {
+        HALT_ERROR(NOT_ALLOWED, "This function can only be called from the context of an initialized debug box.\n\r");
+    }
+
+    /* Reboot.
+     * This can only be called from privileged code. */
+    NVIC_SystemReset();
```

Comment:
If `NVIC_SystemReset` can only be called from privileged code, and `debug_reboot` can only be called from the context of a unprivileged, initialized debug box, why are we calling it?

I think this comment is lying. We can call the function from unprivileged code, but an exception occurs and uVisor traps it and reboots.

Please update the comment to be more clear in explaining what is happening.